### PR TITLE
Defusion: Update jetpack_require_lib calls for class-jetpack-instagram-gallery-helper.php

### PR DESCRIPTION
**Pull request by sdixon194**
#### Changes proposed in this Pull Request:

Updates jetpack_require_lib calls for class-jetpack-instagram-gallery-helper.php 

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/a
#### Does this pull request change what data or activity we track or use?

No
#### Testing instructions:

* Changes make sense?
*

### CODE REVIEW

File: projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram-gallery.php [hunk 1]
```diff
@@ -30,8 +30,8 @@ public function __construct() {
 			}
 		}
 
-		if ( ! class_exists( 'Jetpack_Instagram_Gallery_Helper' ) ) {
-			\jetpack_require_lib( 'class-jetpack-instagram-gallery-helper' );
+		if ( ! class_exists( 'Jetpack_Instagram_Gallery_Helper' ) && defined( 'JETPACK__PLUGIN_DIR' ) ) {
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram-gallery.php]: When would `JETPACK__PLUGIN_DIR` not be defined?

File: projects/plugins/jetpack/extensions/blocks/instagram-gallery/instagram-gallery.php [hunk 2]
```diff
@@ -66,8 +66,8 @@ function render_block( $attributes, $content ) { // phpcs:ignore VariableAnalysi
 		$spacing
 	);
 
-	if ( ! class_exists( 'Jetpack_Instagram_Gallery_Helper' ) ) {
-		\jetpack_require_lib( 'class-jetpack-instagram-gallery-helper' );
+	if ( ! class_exists( 'Jetpack_Instagram_Gallery_Helper' ) && defined( 'JETPACK__PLUGIN_DIR' ) ) {
```
Comment (anomiex) [inline projects/plugins/jetpack/extensions/blocks/instagram-gallery/instagram-gallery.php]: Same

File: projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram-gallery.php (hunk 1 above)
Comment (sdixon194) [reply projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram-gallery.php]: Was just going purely on some precedent when poking around other uses 🤷

**Review by anomiex — DISMISSED** [review]

File: projects/plugins/jetpack/extensions/blocks/instagram-gallery/instagram-gallery.php (hunk 2 above)
Comment (sdixon194) [reply projects/plugins/jetpack/extensions/blocks/instagram-gallery/instagram-gallery.php]: Removed

**Review by anomiex — APPROVED** [review]

**Pull request merged**